terraform/hcl/applicable/req_sw_terraform_aws_api_gateway_encrypt_cache.tf

```diff
@@ -0,0 +1,8 @@
+resource "aws_api_gateway_method_settings" "vulnerable_example" {
+  rest_api_id = "dummy"
+  method_path = "dummy"
+  stage_name = "dummy"
+  settings {
+    caching_enabled = true
+  }
+}
```

Comment:
[comment]: <> (FrogbotReview Iac 1 1 8 2 vulnerable_example /terraform/hcl/applicable/req_sw_terraform_aws_api_gateway_encrypt_cache.tf)

### 🛠️ Infrastructure as Code Vulnerability
	
Severity: ![](https://raw.githubusercontent.com/jfrog/frogbot/master/resources/v2/applicableMediumSeverity.png)<br>  medium

Finding: `cache_data_encrypted=False was detected`

#### 👇 Details

<details>
<summary> <b>Full description</b> </summary>
<br>
Resource `aws_api_gateway_method_settings` should have `settings.cache_data_encrypted=true` (only needed if `settings.caching_enabled=true`)

Vulnerable example - 
```
resource "aws_api_gateway_method_settings" "vulnerable_example" {
  settings {
    caching_enabled = true
  }
}
```

Secure example -
```
resource "aws_api_gateway_method_settings" "secure_example" {
  settings {
    caching_enabled = true
    cache_data_encrypted = true
  }
}
```

</details>

[[🐸 JFrog Frogbot]](https://github.com/jfrog/frogbot#readme)